packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -11,11 +11,38 @@ import 'package:flutter/widgets.dart';
 import 'button_style.dart';
 import 'colors.dart';
 import 'constants.dart';
+import 'elevated_button.dart';
+import 'filled_button.dart';
 import 'ink_well.dart';
 import 'material.dart';
 import 'material_state.dart';
+import 'outlined_button.dart';
+import 'text_button.dart';
 import 'theme_data.dart';
 
+/// {@template flutter.material.ButtonStyleButton.iconAlignment}
+/// Determines the alignment of the icon within the widgets such as:
+///   - [TextButton.icon],
+///   - [ElevatedButton.icon],
+///   - [OutlinedButton.icon],
+///   - [FilledButton.icon],
+///   - [FilledButton.tonalIcon].
+///
+/// The effect of `iconAlignment` depends on [TextDirection]. If textDirection is
+/// [TextDirection.ltr] then [IconAlignment.start] and [IconAlignment.end] align the
+/// icon on the left or right respectively.  If textDirection is [TextDirection.rtl] the
+/// the alignments are reversed.
+///
+/// Defaults to [IconAlignment.start].
```

Comment:
1.  Copy the contents and delete `elevated_button.1.dart` and it's test .
2. Let's add `button_style_button` folder in the `api/lib/material` parent folder.
3. Add the copied contents to a new `icon_aligment.0.dart` example class.
4. Update the examples to showcase all the support buttons in a `OverflowBar` 
5. Replace `DropdownMenu` with `ChoiceChip` in a `OverflowBar` for the icon alignment selection and text direction selection.
6.  Add `button_style_button` in the test folder and test the example selections in both text directions,

Reply:
Since this doc is already a template, the `icon_aligment.0.dart` example will be available in all button docs. Not just in `ElevatedButton`  docs.

Reply:
`DropdownMenu` is a bit much for selection of two items.

`ChoiceChip` should provide easy to access selections

Reply:
> Since this doc is already a template, the `icon_aligment.0.dart` example will be available in all button docs. Not just in `ElevatedButton` docs.

I'll name it button_style_button.icon_alignment.0.dart as guidelines suggest that. https://github.com/flutter/flutter/tree/master/examples/api#naming

Reply:
> 1. Copy the contents and delete `elevated_button.1.dart` and it's test .
> 2. Let's add `button_style_button` folder in the `api/lib/material` parent folder.
> 3. Add the copied contents to a new `icon_aligment.0.dart` example class.
> 4. Update the examples to showcase all the support buttons in a `OverflowBar`
> 5. Replace `DropdownMenu` with `ChoiceChip` in a `OverflowBar` for the icon alignment selection and text direction selection.
> 6. Add `button_style_button` in the test folder and test the example selections in both text directions,



> Since this doc is already a template, the `icon_aligment.0.dart` example will be available in all button docs. Not just in `ElevatedButton` docs.



> `DropdownMenu` is bit much for selection of two items.
> 
> `ChoiceChip` should provide easy to access selections.



> > Since this doc is already a template, the `icon_aligment.0.dart` example will be available in all button docs. Not just in `ElevatedButton` docs.
> 
> I'll name it button_style_button.icon_alignment.0.dart as guidelines suggest that. https://github.com/flutter/flutter/tree/master/examples/api#naming

added the new example with all the suggestions, plus added test. For some reason IconAlignment.end test for FilledButton.tonalIcon is failing in `button_style_button.icon_alignment.0_test.dart` (I've commented its code), will check it in morning. thanks a lot for the suggestions!

Reply:
keeping this one unresolved until I figure the test failure.

Reply:
> keeping this one unresolved until I figure the test failure.

When resolving this, can you please merge commits into a single commit as they're too many right now.

Reply:
Sure I'll squash it to 1.

Reply:
I'll get this sorted by Monday, still unable to catch what's wrong with that particular test. Sorry for the more delay.

Reply:
also, linux-analyzer ci check is failing because example is not linked. I'll fix this too.